plugins/ImageActions/main.lua

```diff
@@ -39,11 +42,97 @@ function cb(mode)
       if image:bands()<4 then
         image = image:bandjoin(255) -- add alpha channel
       end
-      white = {240, 240, 240, 0} -- pixels are considered white, if its r,g,b values are >=240 and alpha is arbitrary
-      transparent = {0, 0, 0, 0}
+      local white = {240, 240, 240, 0} -- pixels are considered white, if its r,g,b values are >=240 and alpha is arbitrary
+      local transparent = {0, 0, 0, 0}
       image = image:more(white):bandand():ifthenelse(transparent, image)
+    elseif mode == 8 then
+      local width = image:width()
+      local height = image:height()
+      image = image:crop(image:find_trim())
+      local newWidth = image:width()
+      local newHeight = image:height()
+      maxWidth, maxHeight = maxWidth * newWidth // width, maxHeight * newHeight // height
+    end
+
+    if mode ~= 9 and mode ~=10 then
+      table.insert(imdata, {data = image:write_to_buffer(".png"), x=x, y=y, maxWidth=maxWidth, maxHeight=maxHeight})
+    else
+      local vertical = mode == 9
+      -- find largest sequence of white or transparent pixel rows/columns
+      local filter = image:colourspace("b-w"):extract_band(0):more(240)
+      local width = filter:width()
+      local height = filter:height()
+      local last = vertical and height or width
+      local isWhite = {}
+
+      for i = 0, last - 1 do
+        if vertical then
+          isWhite[i+1] = filter:extract_area(0, i, width, 1):avg() > 254
+        else
+          isWhite[i+1] = filter:extract_area(i, 0, 1, height):avg() > 254
+        end
+      end
+
+      if #isWhite == 0 then
+        word = vertical and "rows" or "columns"
+        print("There are no completely white " .. word .. " in this image. Skipping.") 
+        table.insert(imdata, {data = im["data"], x=x, y=y, maxHeight=maxHeight, maxWidth=maxWidth})
+        goto continue 
```

Comment:
You've got some trailing whitespaces at some places.

Reply:
I will add a formatting commit.